base/server/python/pki/server/deployment/scriptlets/configuration.py

```diff
@@ -352,7 +352,9 @@ def spawn(self, deployer):
                 secure_port=proxySecurePort,
                 domain_manager=True)
 
-        if not config.str2bool(deployer.mdict['pki_share_db']) and not clone:
+        if config.str2bool(deployer.mdict['pki_ds_user_setup']) and \
+                not config.str2bool(deployer.mdict['pki_share_db']) and \
```

Comment:
I cannot find other use of `pki_share_db` and `pki_ds_user_setup` in this file, or in other. Therefore, we are using 2 bools to control a condition and this should support 4 different scenarios but I do not see them.

@edewata I am missing something or there two options can be merged in one?

Reply:
I was a bit concerned about reusing `pki_share_db` for this purpose since it has different default values for different subsystems:
* for CA it defaults to `False` -> it will create the database user
* for other subsystems it defaults to `True` -> it will use the database user created by CA

but I guess it's fine since the `pki-ca-run` is only used by CA and we'll likely use different scripts for different subsystems so we can specify different values there. I've revised the PR to drop the `pki_ds_user_setup`.

Reply:
I can understand but I think two options for the same thing is not good. Now it is better